modules/creating-a-custom-ingress-controller.adoc

```diff
@@ -0,0 +1,66 @@
+// Module included in the following assemblies:
+//
+// *ingress-controller-dnsmgt.adoc
+
+:_content-type: PROCEDURE
+[id="creating-a-custom-ingress-controller_{context}"]
+= Creating a custom Ingress Controller
+
+As a cluster administrator, you can create a custom Ingress Controller to manage DNS record lifecycle.
+
+.Prerequisites
+
+* Install the OpenShift CLI (`oc`).
+* Log in as a user with `cluster-admin` privileges.
+* DNS record created.
+
+.Procedure
+
+. Create an `IngressController` custom resource to set `dnsManagementPolicy`:
+
++
+[source,terminal]
+----
+oc -n openshift-ingress-operator patch ingresscontrollers/<name> --type=merge --patch='{"spec":{"endpointPublishingStrategy":{"loadBalancer":{"dnsManagementPolicy":"Unmanaged"}}}}'
+----
+
++
+[NOTE]
+====
+Once you update `.spec.endpointPublishingStrategy.loadBalancer.dnsManagementPolicy` field to `Unmanaged`, the existing DNS record must denote the same using the `dnsManagementPolicy` field. You can either retain or delete the DNS record in the cloud provider.
+If the domain on the Ingress Controller does not match with `dnsConfig.Spec.BaseDomain` , the policy is automatically set to `Unmanaged`.
+====
+
++
+.Sample Ingress Controller configuration
++
+[source,yaml]
+----
+apiVersion: operator.openshift.io/v1
+kind: IngressController
+metadata:
+  namespace: openshift-ingress-operator
+  name: <name> <1>
+spec:
+  domain: <domain> <2>
+  endpointPublishingStrategy:
+    type: LoadBalancerService
+    loadBalancer:
+      scope: External <3>
+      dnsManagementPolicy: Unmanaged <4>
+----
+<1> Replace `<name>` with a name for the `IngressController` object.
+<2> Specify the `domain` for the application published by the controller.
+<3> Specify a value of `External` to use an external load balancer.
+<4> `dnsManagementPolicy` indicates if the Ingress Controller is managing the lifecycle of the wildcard DNS record associated with the load balancer.
+By default , the value is `Managed`. The valid values are `Managed` and `Unmanaged`.
+
+
+. Save the file to apply the changes.
++
+[source,terminal]
+----
+oc apply -f <name>.yaml <1>
+----
+
+. Optional: You can delete the associated DNS record in the cloud provider.
```

Comment:
This is only applicable when modifying an Ingress Controller with `Managed` DNS management policy to `Unmanaged`. You can remove it.